src/States.cpp

```diff
@@ -111,6 +111,16 @@ namespace ale {
 
 
   State States::getState(double time, PositionInterpolation interp) const {
+
+
+    // If time is in times, don't need to interpolate!
+    auto candidate_time = std::lower_bound(m_ephemTimes.begin(), m_ephemTimes.end(), time);
+
+    if (*candidate_time == time) {
```

Comment:
lower_bound can return `m_ephemTimes.end()` which you don't want try and access. This if should be `candidate_time != m_ephemTimes.end() && *candidate_time == time` which will short-circuit in that case.

Reply:
Good point, thanks